src/pyhf/constraints.py

```diff
@@ -39,57 +55,70 @@ def _precompute(self):
                 normal_constraint_sigmas.append([1.0] * len(thisauxdata))
 
             normal_constraint_data.append(thisauxdata)
-            normal_constraint_mean_indices.append(self.par_indices[parslice])
-
-        if normal_constraint_mean_indices:
-            normal_mean_idc = default_backend.concatenate(
-                list(
-                    map(
-                        lambda x: default_backend.astensor(x, dtype='int'),
-                        normal_constraint_mean_indices,
-                    )
-                )
-            )
-            normal_sigmas = default_backend.concatenate(
-                list(map(default_backend.astensor, normal_constraint_sigmas))
-            )
-            normal_data = default_backend.concatenate(
-                list(
-                    map(
-                        lambda x: default_backend.astensor(x, dtype='int'),
-                        normal_constraint_data,
-                    )
-                )
+
+        if self.parameter_helper.index_selection:
```

Comment:
when is this false-y?

Reply:
when no constraints of this type are needed, and thus `pnames` is empty

Reply:
please add a comment on that, as this is definitely not clear in the code